src/bin/sail.ml

```diff
@@ -553,11 +563,20 @@ let main () =
 
   feature_check ();
 
+  begin
+    match !opt_require_version with
+    | Some min_version ->
+        if parse_version Manifest.version < parse_version min_version then (
```

Comment:
We should make sure that if `parse_version Manifest.version` fails then we don't fail. There are a few cases where you can build a Sail version that doesn't know its own version (build from source without having git). We should change how the version is detected in the build to avoid that.

Reply:
Hmm how would you want it to behave if you do `--require-version 1.0` but it doesn't know its own version?

Reply:
IMO this should be hard build error and just require the person building to specify the version in some environment variable?

Reply:
Yeah I think that makes sense. I'll update `sail_manifest.ml` to do this.

Reply:
The simplest solution might be to just hardcode the version in `sail.ml`, like `let version = (major, minor, patch)`, and skip all the build logic for detecting it from either opam or git. We could have a check in the CI to make sure the version in the opam file is never out of sync.

Reply:
Ah yeah I like that even more, especially because OPAM does some unreliable Git shenanigans that I don't at all trust.